File: dotnet/Majorsilence.CrystalCmd.NetFrameworkServer/Majorsilence.CrystalCmd.Common/FullReportAnalysisResponse.cs
using System;
using System.Collections.Generic;

namespace Majorsilence.CrystalCmd.Common
{
    public class FullReportAnalysisResponse
    {
        public FullReportAnalysisResponse()
        {
            Parameters = new string[0];
            DataTables = new DataTableAnalysisDto[0];
            SubReports = new FullSubReportAnalysisDto[0];
            ReportObjects = new ReportObjectsDto[0];
        }

        public IEnumerable<string> Parameters { get; set; }
        public IDictionary<string, string> ParametersExtended { get; set; }
        public IEnumerable<DataTableAnalysisDto> DataTables { get; set; }
        public IEnumerable<FullSubReportAnalysisDto> SubReports { get; set; }
        public IEnumerable<ReportObjectsDto> ReportObjects { get; set; }

        public class FullSubReportAnalysisDto
        {
            public FullSubReportAnalysisDto()
            {
                Parameters = new string[0];
                DataTables = new DataTableAnalysisDto[0];
            }

            public string SubreportName { get; set; }
            public IEnumerable<string> Parameters { get; set; }
            public IEnumerable<DataTableAnalysisDto> DataTables { get; set; }
        }

        public class DataTableAnalysisDto
        {
            public DataTableAnalysisDto()
            {
                ColumnNames = new string[0];
            }

            public string DataTableName { get; set; }
            public IEnumerable<string> ColumnNames { get; set; }
        }

        public class ReportObjectsDto
        {
            public string ObjectName { get; set; }
            public int Width { get; set; }
            public int TopPosition { get; set; }
            public string ObjectValue { get; set; }
        }
    }
}

File: java/CrystalCmd/src/FullReportAnalysisResponse.java
import java.util.ArrayList;
import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;

/**
 * Mirrors Majorsilence.CrystalCmd.Common.FullReportAnalysisResponse. Field names are
 * PascalCase to match the JSON the C# server emits, so a C# client can consume a Java
 * server's /analyzer output and vice versa.
 */
public class FullReportAnalysisResponse {
    public List<String> Parameters = new ArrayList<>();
    public Map<String, String> ParametersExtended = new LinkedHashMap<>();
    public List<DataTableAnalysisDto> DataTables = new ArrayList<>();
    public List<FullSubReportAnalysisDto> SubReports = new ArrayList<>();
    public List<ReportObjectsDto> ReportObjects = new ArrayList<>();

    public static class FullSubReportAnalysisDto {
        public String SubreportName;
        public List<String> Parameters = new ArrayList<>();
        public List<DataTableAnalysisDto> DataTables = new ArrayList<>();
    }

    public static class DataTableAnalysisDto {
        public String DataTableName;
        public List<String> ColumnNames = new ArrayList<>();
    }

    public static class ReportObjectsDto {
        public String ObjectName;
        public int Width;
        public int TopPosition;
        public String ObjectValue;
    }
}
